### DISCUSSION TITLE
CORS error

### DISCUSSION THREAD

**Discussion by anshuljain633** [miguelgrinberg/Flask-SocketIO#1590] [Q&A]
Hi @miguelgrinberg 
I'm getting this error at the client:
```
Access to XMLHttpRequest at 'http://x.com/socket.io/?EIO=4&transport=polling&t=NVyXb_5' from origin 'http://x.com' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.

polling-xhr.js:202 GET http://x.com/socket.io/?EIO=4&transport=polling&t=NVyXb_5 net::ERR_FAILED
```

i'm using supervisor to run gunicorn server
```
[program:gunicorn]
command=gunicorn --bind 0.0.0.0:5000 -k flask_sockets.worker app:app

```
client side code

> 
<script src="https://cdnjs.cloudflare.com/ajax/libs/socket.io/3.1.1/socket.io.js"></script>
        <script type="text/javascript">
                $(document).ready(function(){

                        // sending a connect request to the server.
                        var socket = io.connect('https://x.com');

                        // An event handler for a change of value
                        $('input.sync').on('input', function(event) {
                socket.emit('Slider value changed', {who: $(this).attr('id'), data: $(this).val()});
                return false;
            });

            socket.on('after connect', function(msg){
                console.log('After connect', msg);
            });

            socket.on('update value', function(msg) {
                console.log('Slider value updated');
                $('#'+msg.who).val(msg.data);
            });
                });
        </script>

server side -> 
> from flask import Flask, render_template
from flask_socketio import SocketIO, emit
from flask.ext.cors import CORS

app = Flask(__name__)
socketio = SocketIO(app,cors_allowed_origins="*")

values = {
    'slider1': 25,
    'slider2': 0,
}

@app.route('/')
def index():
    return render_template('index.html',**values)

@socketio.on('connect')
def test_connect():
    emit('after connect',  {'data':'Lets dance'})

@socketio.on('Slider value changed')
def value_changed(message):
    values[message['who']] = message['data']
    emit('update value', message, broadcast=True)

cors = CORS(app,resources={r"/*":{"origins":"*"}})
if __name__ == '__main__':
    socketio.run(app, host='0.0.0.0')

**Comment by miguelgrinberg** [comment]
The server configuration seems okay to me. Some intermediary that you have between your server and your client might be stripping the CORS headers from the responses.

**Comment by anshuljain633** [comment]
I'm using nginx container and supervisor to run gunicorn server.
everything works fine if i run application without application server(gunicorn or uwsgi) but when i configure any of application server i get this CORS error.
beside that i'm also using AWS ELB.
@miguelgrinberg

**Reply by selmanbaskaya** [reply]
@anshuljain633 Hi mate. 

I am getting exactly the same error as you. 
Did you solve this problem?
